remote-wallet/src/locator.rs

```diff
@@ -0,0 +1,628 @@
+        ));
+        let bad_path = "bad-derivation-path".to_string();
+        assert!(matches!(
+            Locator::new_from_parts(manufacturer, None::<Pubkey>, Some(bad_path.as_str())),
+            Err(LocatorError::DerivationPathError(
+                DerivationPathError::InvalidDerivationPath(_)
+            )),
+        ));
+    }
+
+    #[test]
+    fn test_locator_new_from_uri() {
+        let derivation_path = DerivationPath::new_bip44(Some(0), Some(0));
+        let manufacturer = Manufacturer::Ledger;
+        let pubkey = Pubkey::new_unique();
+        let pubkey_str = pubkey.to_string();
+
+        // usb://ledger/{PUBKEY}?key=0'/0'
+        let mut builder = URIReferenceBuilder::new();
+        builder
+            .try_scheme(Some("usb"))
+            .unwrap()
+            .try_authority(Some(Manufacturer::Ledger.as_ref()))
+            .unwrap()
+            .try_path(pubkey_str.as_str())
+            .unwrap()
+            .try_query(Some("key=0/0"))
+            .unwrap();
+        let uri = builder.build().unwrap();
+        let expect = Locator {
+            manufacturer,
+            pubkey: Some(pubkey),
+            derivation_path: Some(derivation_path.clone()),
+        };
+        assert_eq!(Locator::new_from_uri(&uri), Ok(expect));
+
+        // usb://ledger/{PUBKEY}
+        let mut builder = URIReferenceBuilder::new();
+        builder
+            .try_scheme(Some("usb"))
+            .unwrap()
+            .try_authority(Some(Manufacturer::Ledger.as_ref()))
+            .unwrap()
+            .try_path(pubkey_str.as_str())
+            .unwrap();
+        let uri = builder.build().unwrap();
+        let expect = Locator {
+            manufacturer,
+            pubkey: Some(pubkey),
+            derivation_path: None,
+        };
+        assert_eq!(Locator::new_from_uri(&uri), Ok(expect));
+
+        // usb://ledger
+        let mut builder = URIReferenceBuilder::new();
+        builder
+            .try_scheme(Some("usb"))
+            .unwrap()
+            .try_authority(Some(Manufacturer::Ledger.as_ref()))
+            .unwrap()
+            .try_path("")
+            .unwrap();
+        let uri = builder.build().unwrap();
+        let expect = Locator {
+            manufacturer,
+            pubkey: None,
+            derivation_path: None,
+        };
+        assert_eq!(Locator::new_from_uri(&uri), Ok(expect));
+
+        // usb://ledger/
+        let mut builder = URIReferenceBuilder::new();
+        builder
+            .try_scheme(Some("usb"))
+            .unwrap()
+            .try_authority(Some(Manufacturer::Ledger.as_ref()))
+            .unwrap()
+            .try_path("/")
+            .unwrap();
+        let uri = builder.build().unwrap();
+        let expect = Locator {
+            manufacturer,
+            pubkey: None,
+            derivation_path: None,
+        };
+        assert_eq!(Locator::new_from_uri(&uri), Ok(expect));
+
+        // usb://ledger?key=0'/0'
```

Comment:
```suggestion
        // usb://ledger?key=0/0
```
nit

Reply:
I added a separate instance here for both forms (with consistent comment!)